src/west/commands/project.py

```diff
@@ -850,6 +851,12 @@ def _update_west(rebase, keep_descendants):
         old_sha = _sha(project, 'HEAD')
         _update(project, rebase, keep_descendants)
 
+        if config.get('zephyr', 'zephyr-base', fallback=None) is None:
```

Comment:
I'm confused about why this is in `_update_west()`; it seems like a strange place to put it. What am I missing?

I would have imagined that this could be done in `set_zephyr_base()` in main.py. That way the configuration option, if missing, would get set before a `west build` needs it, for example.

Reply:
as described here: https://github.com/zephyrproject-rtos/west/pull/211#discussion_r264314187
in future, this should not be needed to test, as the value should always be there, but for time being, there are installations out there, where the value is not present, and for those we can simply set the value during part of `selfupdate` and then in future remove this code.

Reply:
Resolving this thread since we have another one open for it